File: whatsapp_extension/icons/generate-icons.js
// Simple icon generator using canvas
// To use: Open generate-icons.html in a browser and click "Download All Icons"
// Or use this Node.js script if you have canvas installed

const fs = require('fs');
const { createCanvas } = require('canvas');

function drawIcon(size) {
    const canvas = createCanvas(size, size);
    const ctx = canvas.getContext('2d');
    const scale = size / 128;
    
    ctx.scale(scale, scale);
    
    // Background
    ctx.fillStyle = '#1a73e8';
    ctx.fillRect(0, 0, 128, 128);
    
    ctx.translate(24, 24);
    
    // Signal waves
    ctx.strokeStyle = '#fff';
    ctx.lineWidth = 4;
    ctx.lineCap = 'round';
    
    ctx.beginPath();
    ctx.moveTo(40, 10);
    ctx.quadraticCurveTo(30, 20, 40, 30);
    ctx.stroke();
    
    ctx.beginPath();
    ctx.moveTo(40, 10);
    ctx.quadraticCurveTo(50, 20, 40, 30);
    ctx.stroke();
    
    ctx.globalAlpha = 0.7;
    ctx.lineWidth = 3;
    
    ctx.beginPath();
    ctx.moveTo(40, 5);
    ctx.quadraticCurveTo(25, 20, 40, 35);
    ctx.stroke();
    
    ctx.beginPath();
    ctx.moveTo(40, 5);
    ctx.quadraticCurveTo(55, 20, 40, 35);
    ctx.stroke();
    
    ctx.globalAlpha = 1.0;
    
    // Center dot
    ctx.fillStyle = '#fff';
    ctx.beginPath();
    ctx.arc(40, 20, 4, 0, Math.PI * 2);
    ctx.fill();
    
    // Connector line
    ctx.strokeStyle = '#fff';
    ctx.lineWidth = 3;
    ctx.beginPath();
    ctx.moveTo(40, 24);
    ctx.lineTo(40, 45);
    ctx.stroke();
    
    // Network nodes
    ctx.fillStyle = '#fff';
    ctx.beginPath();
    ctx.arc(20, 55, 5, 0, Math.PI * 2);
    ctx.fill();
    
    ctx.beginPath();
    ctx.arc(40, 50, 5, 0, Math.PI * 2);
    ctx.fill();
    
    ctx.beginPath();
    ctx.arc(60, 55, 5, 0, Math.PI * 2);
    ctx.fill();
    
    // Connection lines
    ctx.strokeStyle = '#fff';
    ctx.lineWidth = 2;
    ctx.beginPath();
    ctx.moveTo(25, 55);
    ctx.lineTo(35, 50);
    ctx.stroke();
    
    ctx.beginPath();
    ctx.moveTo(45, 50);
    ctx.lineTo(55, 55);
    ctx.stroke();
    
    // Data packets
    ctx.fillStyle = '#4CAF50';
    ctx.fillRect(36, 60, 8, 8);
    
    ctx.globalAlpha = 0.8;
    ctx.fillRect(16, 65, 6, 6);
    ctx.fillRect(58, 65, 6, 6);
    
    return canvas;
}

// Generate icons
const sizes = [16, 48, 128];
sizes.forEach(size => {
    const canvas = drawIcon(size);
    const buffer = canvas.toBuffer('image/png');
    fs.writeFileSync(`icon${size}.png`, buffer);
    console.log(`Generated icon${size}.png`);
});

console.log('All icons generated successfully!');

File: whatsapp_extension/icons/generate-icons.py
#!/usr/bin/env python3
"""
Simple icon generator for cURL & WS Capture extension
Requires: pip install pillow
"""

try:
    from PIL import Image, ImageDraw
    import os
except ImportError:
    print("Please install Pillow: pip install pillow")
    exit(1)

def draw_icon(size):
    """Draw the extension icon at specified size"""
    # Create image with transparent background
    img = Image.new('RGBA', (size, size), (0, 0, 0, 0))
    draw = ImageDraw.Draw(img)
    
    scale = size / 128.0
    
    # Background - blue rounded rectangle
    bg_color = (26, 115, 232, 255)  # #1a73e8
    draw.rounded_rectangle([(0, 0), (size-1, size-1)], 
                          radius=int(20*scale), 
                          fill=bg_color)
    
    # Scale coordinates
    def s(coord):
        """Scale a single coordinate"""
        return int(coord * scale)
    
    def sp(x, y):
        """Scale a point (x, y)"""
        return (s(x), s(y))
    
    white = (255, 255, 255, 255)
    green = (76, 175, 80, 255)
    
    # Center area offset
    offset_x = s(24)
    offset_y = s(24)
    
    # Draw signal waves (simplified as arcs)
    # Center dot
    center_x = offset_x + s(40)
    center_y = offset_y + s(20)
    dot_radius = s(4)
    draw.ellipse([center_x - dot_radius, center_y - dot_radius,
                  center_x + dot_radius, center_y + dot_radius],
                 fill=white)
    
    # Connector line
    line_start_y = center_y + dot_radius
    line_end_y = offset_y + s(45)
    line_width = max(1, s(3))
    draw.line([center_x, line_start_y, center_x, line_end_y],
              fill=white, width=line_width)
    
    # Network nodes
    nodes = [(20, 55), (40, 50), (60, 55)]
    node_radius = s(5)
    for nx, ny in nodes:
        node_x = offset_x + s(nx)
        node_y = offset_y + s(ny)
        draw.ellipse([node_x - node_radius, node_y - node_radius,
                      node_x + node_radius, node_y + node_radius],
                     fill=white)
    
    # Connection lines between nodes
    node1_x = offset_x + s(25)
    node1_y = offset_y + s(55)
    node2_x = offset_x + s(35)
    node2_y = offset_y + s(50)
    draw.line([node1_x, node1_y, node2_x, node2_y],
              fill=white, width=max(1, s(2)))
    
    node3_x = offset_x + s(45)
    node3_y = offset_y + s(50)
    node4_x = offset_x + s(55)
    node4_y = offset_y + s(55)
    draw.line([node3_x, node3_y, node4_x, node4_y],
              fill=white, width=max(1, s(2)))
    
    # Data packets (rectangles)
    packet_size = s(8)
    packet_x = offset_x + s(36)
    packet_y = offset_y + s(60)
    draw.rectangle([packet_x, packet_y, 
                   packet_x + packet_size, packet_y + packet_size],
                  fill=green)
    
    # Smaller packets
    small_packet = s(6)
    packet2_x = offset_x + s(16)
    packet2_y = offset_y + s(65)
    draw.rectangle([packet2_x, packet2_y,
                   packet2_x + small_packet, packet2_y + small_packet],
                  fill=green)
    
    packet3_x = offset_x + s(58)
    packet3_y = offset_y + s(65)
    draw.rectangle([packet3_x, packet3_y,
                   packet3_x + small_packet, packet3_y + small_packet],
                  fill=green)
    
    # Draw curved signal waves (simplified)
    wave_color = (255, 255, 255, 180)
    wave_width = max(1, s(3))
    
    # Inner waves
    draw.arc([center_x - s(15), center_y - s(15),
             center_x + s(15), center_y + s(15)],
            start=45, end=135, fill=white, width=s(4))
    draw.arc([center_x - s(15), center_y - s(15),
             center_x + s(15), center_y + s(15)],
            start=-45, end=45, fill=white, width=s(4))
    
    return img

def main():
    """Generate icons in different sizes"""
    sizes = [16, 48, 128]
    
    script_dir = os.path.dirname(os.path.abspath(__file__))
    
    for size in sizes:
        img = draw_icon(size)
        filename = os.path.join(script_dir, f'icon{size}.png')
        img.save(filename, 'PNG')
        print(f'✓ Generated {filename}')
    
    print('\n✓ All icons generated successfully!')
    print('\nThe icons are ready to use in your Chrome extension.')

if __name__ == '__main__':
    main()
